### PR TITLE
Made `Object.hashAll` and `Object.hashAllUnordered` accept nullable lists

### LINKED ISSUE #48094: Make `Object.hashAll` and `Object.hashAllUnordered` accept nullable lists
Unlike `Object.hash`, `Object.hashAll` and `Object.hashAllUnordered` does not accept nullable parameters, which it should to make it consistent with `Object.hash` and flutter's (soon-to-be previous) hash function (`hashList`).

### PR DESCRIPTION
**Pull request by werainkhatri**
Fixes #48094

### CODE REVIEW

**Comment by mit-mit** [comment]
Review in https://dart-review.googlesource.com/c/sdk/+/226806

File: sdk/lib/core/object.dart [hunk 1]
```diff
@@ -508,7 +508,10 @@ class Object {
   /// or between different versions of the platform libraries,
   /// and it may depend on values that change on each program execution.
   @Since("2.14")
-  static int hashAll(Iterable<Object?> objects) {
+  static int hashAll(Iterable<Object?>? objects) {
+    if(objects == null) {
```
Comment (mateusfccp) [inline sdk/lib/core/object.dart:512]:
```suggestion
    if (objects == null) {
```

File: sdk/lib/core/object.dart [hunk 2]
```diff
@@ -546,7 +549,10 @@ class Object {
   /// or between different versions of the platform libraries,
   /// and it may depend on values that change per program run
   @Since("2.14")
-  static int hashAllUnordered(Iterable<Object?> objects) {
+  static int hashAllUnordered(Iterable<Object?>? objects) {
+    if(objects == null) {
```
Comment (mateusfccp) [inline sdk/lib/core/object.dart:553]:
```suggestion
    if (objects == null) {
```

**Comment by mit-mit** [comment]
Closing as per review comments in https://dart-review.googlesource.com/c/sdk/+/226806

**Pull request closed without merging**